src/App.tsx

```diff
@@ -1,19 +1,134 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
 import { AuthContext } from './components/Auth/AuthContext';
+import { TodoList } from './components/TodoList';
+import { FilterList } from './components/FilterList';
+import { getTodos, pushTodo } from './api/todos';
+import { Todo } from './types/Todo';
+import { ErrorNotification } from './components/ErrorNotification';
+import { Filter } from './types/Filter';
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
+  const [todoList, setTodoList] = useState<Todo[]>([]);
+  const [visibleTodoList, setVisibleList] = useState<Todo[]>([]);
+  const [newTitle, setNewTitle] = useState('');
+
+  const [isAdding, setIsAdding] = useState(false);
+
+  const [emptyFieldError, setEmptyError] = useState(false);
+  const [failedAddError, setAddError] = useState(false);
+  const [failedDeleteError, setDeleteError] = useState(false);
+
+  const findTodos = async () => {
+    const loadedTodos = await getTodos(user?.id) || null;
+
+    setTodoList(loadedTodos);
+    setVisibleList(loadedTodos);
+  };
 
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
     if (newTodoField.current) {
       newTodoField.current.focus();
     }
+
+    findTodos();
   }, []);
 
+  const cancelErrors = () => {
+    setEmptyError(false);
+    setAddError(false);
+  };
+
+  const findNewTodoId = (): number => {
+    if (todoList.length !== 0) {
+      const max = todoList.reduce((a, b) => (a.id > b.id ? a : b));
+
+      return max.id + 1;
+    }
+
+    return 1;
+  };
+
+  const pushTodos = async () => {
+    const pushedTodo = await pushTodo(findNewTodoId(), newTitle, user?.id);
+
+    setVisibleList(prevTodos => {
+      return [...prevTodos, pushedTodo];
+    });
+    setTodoList(prevTodos => {
+      return [...prevTodos, pushedTodo];
+    });
+  };
+
+  const deleteTodos = (id: number) => {
+    try {
+      setVisibleList(prevTodos => {
+        return (prevTodos.filter(item => item.id !== id));
+      });
+      setTodoList(prevTodos => {
+        return (prevTodos.filter(item => item.id !== id));
+      });
+    } catch {
+      setDeleteError(true);
+    }
+  };
+
+  const countActiveItems = (): number => {
+    let sum = 0;
+
+    todoList.forEach(item => {
+      if (!item.completed) {
+        sum += 1;
+      }
+
+      return sum;
+    });
+
+    return sum;
+  };
+
+  const filterTodos = async (filterBy: Filter) => {
+    setVisibleList(todoList);
```

Comment:
you don't need `setVisibleList` here. if you do it in cases it will be the same. just use default case correctly